### PR TITLE
extract-tests: remove unused test scripts and devDeps from addon

### PR DESCRIPTION
**Pull request by simonihmig**
As the addon does not contain any tests anymore, we can remove package.json scripts and devDependencies that relate to testing.

### CODE REVIEW

**Comment by simonihmig** [comment]
@NullVoxPopuli any idea what could cause the failing test here? I tried this locally, by running extract-tests on `base-ts-v1-addon`, and wasn't able to reproduce. That same `tsc` was working fine. And `@types/test-helpers` isn't even in my `node_modules`. But it seems to be there for the CI run, if I am able to understand the error message correctly? 🤔

**Comment by NullVoxPopuli** [comment]
I've seen that happen before locally -- and it comes from out of date dependencies -- because `@ember/test-helpers` ships its own types now, and we don`t need `@types/ember__test-helpers`
It _could be_ that the fixture for the base-ts-v1-addon needs to have its dependencies bumped just for the test to be stable -- but it's a little obnoxious -- I kind of blame DefinitelyTyped for this though, because they use `*` for all their versions, it means _anything_ is valid, even if out of date -- so hoops need to be jumped through to get the correct versions of those deps

File: cli/src/extract-tests/index.js [hunk 1]
```diff
@@ -250,4 +263,22 @@ async function deleteExtraAddonFiles(analysis) {
     force: true,
     recursive: true,
   });
+
+  await packageJson.modify((json) => {
```
Comment (NullVoxPopuli) [inline cli/src/extract-tests/index.js:267]: there is also:
 - https://ember-apply.pages.dev/funcs/packageJson.removeDevDependencies
 - https://ember-apply.pages.dev/funcs/packageJson.removeDependencies
 
 but no remove scripts :upside_down_face: probably fine. and what you have is fine, too
 
 :100:

File: cli/src/extract-tests/index.js [hunk 2]
```diff
@@ -250,4 +263,22 @@ async function deleteExtraAddonFiles(analysis) {
...
+    // Remove any test-related scripts, as they now will be in the test-app
+    json.scripts = Object.fromEntries(
+      Object.entries(json.scripts).filter(
+        ([key]) => key !== 'test' && key !== '_test' && !key.match(/^test:.*/)
+      )
```
Comment (NullVoxPopuli) [inline cli/src/extract-tests/index.js:272]:
```suggestion
      Object.entries(json.scripts).filter(
        ([key]) => !key.includes('test')
      )
```

File: cli/src/extract-tests/index.js [hunk 3]
```diff
@@ -250,4 +263,22 @@ async function deleteExtraAddonFiles(analysis) {
...
+    );
+
+    // Remove testing dependencies from devDependencies, unless they are in peerDependencies, which means the addon's own code (e.g. /addon-test-support) might need them
```
Comment (NullVoxPopuli) [inline cli/src/extract-tests/index.js:275]: thanks for the comment!!! This is a helpful / needed consideration

File: cli/src/extract-tests/index.js (hunk 1 above)
Comment (simonihmig) [reply cli/src/extract-tests/index.js:267]: Yeah, I know, just thought that given I already do `.modify()`, I can do everything in that place. But I have no string feelings about that...

File: cli/src/extract-tests/index.js (hunk 2 above)
Comment (simonihmig) [reply cli/src/extract-tests/index.js:272]: Yeah, I wanted to do this first, but then thought maybe this is too broad and could include some unwanted changes? But happy to change if you feel comfortable doing so!

**Comment by simonihmig** [comment]
> the fixture for the base-ts-v1-addon needs to have its dependencies bumped just for the test to be stable

Yeah, I just removed `@types/ember__test-helpers` from the fixture, and it fixes the test! 

Still weird, because `extract-tests` will do this also [here](https://github.com/NullVoxPopuli/ember-addon-migrator/pull/77/files#diff-4b139fca6d2f2e67bb0e240c551eba333413a72ce4313e8fb10516fbdf474dceR25), but still somehow that dependency apparently sneaks into `node_modules` somehow? The test-app still has it (until we land the `--typescript` blueprint refactoring finally), but when using pnpm this shouldn't affect the addon workspace, right?

So many weird things going on sometimes. But anyway, I think the change itself here should be ok, so I'm fine with that workaround to make tests happy if you are...

File: cli/src/extract-tests/index.js (hunk 2 above)
Comment (NullVoxPopuli) [reply cli/src/extract-tests/index.js:272]: yeah, I think it's safeish.
The main thing is that it's less logic, and doesn't include the CrowdStrike convention of the leading _.
(I use `_:` in in some other project, for example, so matching on just `test` inclusion would cover all of this)

File: cli/src/extract-tests/index.js (hunk 1 above)
Comment (NullVoxPopuli) [reply cli/src/extract-tests/index.js:267]: I don't either -- I left this comment before I got to the end of the function, and what exists as you wrote it is pretty important, so I'm happy

**Review by NullVoxPopuli — APPROVED** [review]

**Pull request merged**